src/itehandler/IteToSwitch.cc

```diff
@@ -233,6 +233,12 @@ ite::Dag IteToSwitch::constructIteDag(PTRef root, Logic const & logic) {
 
         } else { // not Ite
             for (PTRef child : t) {
+                if (logic.hasIntegers() || logic.hasReals()) {
+                    if (logic.isNonlin(child)) {
+                        auto termStr = logic.pp(child);
+                        throw ApiException("Nonlinear operations in the formula: " + termStr);
+                    }
+                }
```

Comment:
I don't think we need a check here.
ITE manipulation should be agnostic of arithmetic operations.

Reply:
But ideally it would be a check in single place
My idea is to put it in the code where we insert assertions into the code somewhere...
Do you have some ideas where better to put it?

Reply:
Yes, it should be in one place.
We can put the check when we get the assertions, but I think better place would be to put the check in LASolver.
If we start supporting nonlinear arithmetic, that would be the place to change.

Reply:
Moved, no checks in IteToSwitch anymore